wgengine/router/osrouter tests segfaulting with Linux platform on main

Tests for `wgengine/router/osrouter`, viz `TestDelRouteIdempotent` and `TestAddRemoveRules`, are segfaulting on my linux box @ 03c3551ee50921cd4dba46e8fe08a6a49b0e9920

Example:

```
$ uname -a
Linux ...snip... 6.17.0-1009-aws #9~24.04.2-Ubuntu SMP Fri Mar  6 23:50:29 UTC 2026 x86_64 x86_64 x86_64 GNU/Linux
$ ./tool/go test -v ./wgengine/router/osrouter -run TestDelRouteIdempotent
=== RUN   TestDelRouteIdempotent
--- FAIL: TestDelRouteIdempotent (0.01s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0xe8 pc=0x9232ef]

goroutine 8 [running]:
testing.tRunner.func1.2({0xa20300, 0x1026370})
        /src/testing/testing.go:1974 +0x232
testing.tRunner.func1()
        ...snip.../src/testing/testing.go:1977 +0x349
panic({0xa20300?, 0x1026370?})
        ...snip.../src/runtime/panic.go:860 +0x13a
tailscale.com/wgengine/router/osrouter.(*linuxRouter).getV6Available(...)
        ...snip.../tailscale/wgengine/router/osrouter/router_linux.go:786
tailscale.com/wgengine/router/osrouter.(*linuxRouter).addRoute(0x384c7cc9c000, {{{0xc?, 0xf?}, {0x384c7c8a2108?}}, 0x51?})
        ...snip.../tailscale/wgengine/router/osrouter/router_linux.go:877 +0x4f
tailscale.com/wgengine/router/osrouter.TestDelRouteIdempotent(0x384c7c9626c8)
        ...snip.../tailscale/wgengine/router/osrouter/router_linux_test.go:1222 +0x11e
testing.tRunner(0x384c7c9626c8, 0xb1e780)
        ...snip.../src/testing/testing.go:2036 +0xea
created by testing.(*T).Run in goroutine 1
        ...snip.../src/testing/testing.go:2101 +0x4c5
FAIL    tailscale.com/wgengine/router/osrouter  0.017s
FAIL
```

--- a/wgengine/router/osrouter/router_linux_test.go
+++ b/wgengine/router/osrouter/router_linux_test.go
@@ -1211,7 +1211,9 @@ func TestRuleDeletedEvent(t *testing.T) {
 }
 
 func TestDelRouteIdempotent(t *testing.T) {
+	fake := NewFakeOS(t)
 	lt, _ := newLinuxRootTest(t)
+	lt.r.nfr = fake.nfr
 	defer lt.Close()
 
 	for _, s := range []string{
@@ -1237,7 +1239,9 @@ func TestDelRouteIdempotent(t *testing.T) {
 }
 
 func TestAddRemoveRules(t *testing.T) {
+	fake := NewFakeOS(t)
 	lt, _ := newLinuxRootTest(t)
+	lt.r.nfr = fake.nfr
 	defer lt.Close()
 	r := lt.r
 

